pycbc/events/single.py

```diff
@@ -1,61 +1,141 @@
                  duration_threshold=0,
-                 fixed_ifar=0):
+                 fit_bins=None,
+                 fit_coeffs=None,
+                 fit_rates=None,
+                 fixed_ifar=None,
+                 fit_thresh=None):
         self.ifo = ifo
-        self.newsnr_threshold = newsnr_threshold
-        self.reduced_chisq_threshold = reduced_chisq_threshold
-        self.fixed_ifar = fixed_ifar
-        self.duration_threshold = duration_threshold
+        self.thresholds = {
+            "newsnr": newsnr_threshold,
+            "reduced_chisq": reduced_chisq_threshold,
+            "duration": duration_threshold}
+        self.fit_info = {
+            "fixed_ifar": fixed_ifar,
+            "bins": fit_bins,
+            "rates": fit_rates,
+            "coeffs": fit_coeffs,
+            "thresh": fit_thresh}
 
     @staticmethod
     def insert_args(parser):
         parser.add_argument('--single-newsnr-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
         parser.add_argument('--single-reduced-chisq-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
-        parser.add_argument('--single-fixed-ifar', nargs='+',
+        parser.add_argument('--single-fixed-ifar', nargs='+', default=None,
                             type=float, action=MultiDetOptionAction)
+        parser.add_argument('--single-fit-file',
+                            help="File which contains definitons of fit "
+                                 "coefficients and counts for specific "
+                                 "single trigger IFAR fitting.")
+        parser.add_argument('--sngl-ifar-est-dist', nargs='+',
+                            default='conservative',
+                            action=MultiDetOptionAction)
         parser.add_argument('--single-duration-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
 
     @classmethod
     def from_cli(cls, args, ifo):
+        if args.sngl_ifar_est_dist[ifo] not in \
+                ['conservative', 'mean', 'fixed']:
+            raise NotImplementedError("Single fitting distribution must be "
+                                      "conservative, mean or fixed")
+        if args.sngl_ifar_est_dist[ifo] == "fixed":
+            fit_bins = None
+            fit_rates = None
+            fit_thresh = None
+            fit_coeffs = None
+        else:
+            with h5py.File(args.single_fit_file, 'r') as fit_file:
+                bin_edges = fit_file['bins_edges'][:]
+                fit_bins = bin_utils.IrregularBins(bin_edges)
+                dist_grp = fit_file[ifo + '/' + args.sngl_ifar_est_dist[ifo]]
+                live_time = fit_file[ifo].attrs['live_time']
+                fit_rates = dist_grp['counts'][:] / live_time
+                fit_coeffs = dist_grp['fit_coeff'][:]
+                fit_thresh = fit_file.attrs['fit_threshold']
```

Comment:
You could also avoid passing many arguments to the class constructor by having the constructor itself read the HDF5 file, I guess (i.e. only pass the file name to the constructor). But this looks good enough if codeclimate is happy.

Reply:
Updated to do this so that fewer args are passed through the cli constructor